src/osx/cocoa/textctrl.mm

```diff
@@ -1114,19 +1325,33 @@ - (void)controlTextDidEndEditing:(NSNotification *)aNotification
     if ( !m_textView )
         return;
 
-    if ( start == -1 && end == -1 )
+    NSTextStorage* storage = [m_textView textStorage];
+
+    NSRange range = NSMakeRange(start, end-start);
+    if ( ![m_textView shouldChangeTextInRange:range replacementString:nil] ) // for Undo
+        return;
+
+    @try
     {
-        NSMutableDictionary* const
-            attrs = [NSMutableDictionary dictionaryWithCapacity:5];
-        if ( style.HasFont() )
-            [attrs setValue:style.GetFont().OSXGetNSFont() forKey:NSFontAttributeName];
-        if ( style.HasBackgroundColour() )
-            [attrs setValue:style.GetBackgroundColour().OSXGetNSColor() forKey:NSBackgroundColorAttributeName];
-        if ( style.HasTextColour() )
-            [attrs setValue:style.GetTextColour().OSXGetNSColor() forKey:NSForegroundColorAttributeName];
-        if ( style.HasFontUnderlined() )
+        int textsEffects = style.GetTextEffects();
+
+        if ( style.IsCharacterStyle() )
         {
-            int underlineStyle = NSUnderlineStyleNone;
+            // It's not obvious whether strikethrough is denoted
+            // definitively by wxTEXT_ATTR_EFFECT_STRIKETHROUGH
+            // or by wxTEXT_ATTR_FONT_STRIKETHROUGH, so handle both.
+            int strikeThroughStyle = NSUnderlineStyleNone; // should be type NSUnderlineStyle
```

Comment:
Why isn't it declared as being of the proper type?

Reply:
See discussion on wx-dev.  Declaring it as NSUnderlineStyle directly fails the xcode6.4 build test.